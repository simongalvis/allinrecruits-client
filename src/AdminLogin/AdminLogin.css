.AdminLogin{
    height:100vh;
    display: flex;
    flex-direction: column;
    background-color: rgb(75, 75, 75);
    
}
#admin-login-heading{
    background-color: white;
    margin-bottom: 0;
    padding-bottom: 17px;
}

#admin-login-button{
    margin: 24px 24px;
    width: 177px;
    height:33px;
    font-size: 17px;
    border-radius: 33px;
    background-color: gray;
    color: white;
    border: 1px solid black;
}