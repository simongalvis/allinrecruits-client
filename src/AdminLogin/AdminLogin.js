import React from 'react';
import { Link } from 'react-router-dom'
import './AdminLogin.css'
import NavBar from '../NavBar/NavBar'
import ApiContext from '../ApiContext';

class AdminLogin extends React.Component{
    static contextType = ApiContext;

    handleSubmit = e =>{
        e.preventDefault();
       
    
    const  { loginUsername, loginPassword } = e.target;
    this.context.loginAdmin(loginUsername.value, loginPassword.value )
       
      
       
      setTimeout( () => {if (Object.keys(this.context.loggedAdmin).length > 0){
               this.props.history.push('/admin-dashboard')}}, 1000) 
        
       
    
    
    
    
    }


    render(){
        return(
            <div className="AdminLogin">
                <NavBar/>
                <h1>AdminLogin</h1>
                <form onSubmit={this.handleSubmit}>
                    <label htmlFor="loginUsername"><b>Username:</b> </label>
                    <input type="text" id="loginUsername" name="loginUsername" autoCorrect="off" autoCapitalize="none" required/>
                    <label htmlFor="loginPassword"><b>Password:</b> </label>
                    <input type="password" id="loginPassword" name="loginPassword" autoCorrect="off" autoCapitalize="none" required/>
                    
                    <button type="submit">Log In</button>
                    
                </form>
            </div>
        )
    }
}
export default AdminLogin;