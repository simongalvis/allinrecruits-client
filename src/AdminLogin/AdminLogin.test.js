import React from 'react'
import ReactDOM from 'react-dom'
import AdminLogin from './AdminLogin'
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'

describe(`AdminLogin component`, () =>{
    it(`renders without crashing`, () =>{
        const div = document.createElement('div');

        ReactDOM.render(<BrowserRouter><AdminLogin/></BrowserRouter>, div)

        ReactDOM.unmountComponentAtNode(div);
    })
    it(`Renders UI as expected`, () =>{
        const tree = renderer
            .create(<BrowserRouter><AdminLogin/></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})