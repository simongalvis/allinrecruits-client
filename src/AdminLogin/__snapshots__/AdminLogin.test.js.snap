// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminLogin component Renders UI as expected 1`] = `
<div
  className="AdminLogin"
>
  <div
    className="NavBar"
  >
    <header>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="all-in-education-logo"
          id="main-logo"
          src="all-in-logo.png"
        />
      </a>
      <a
        href="/admin-login"
      >
        <button
          id="admin-login-btn"
        >
          Admin Login
        </button>
      </a>
    </header>
  </div>
  <h1
    id="admin-login-heading"
  >
    Administrator Login
  </h1>
  <form
    id="admin-login-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="loginUsername"
    >
      <b>
        Username:
      </b>
       
    </label>
    <input
      autoCapitalize="none"
      autoCorrect="off"
      id="loginUsername"
      name="loginUsername"
      required={true}
      type="text"
    />
    <label
      htmlFor="loginPassword"
    >
      <b>
        Password:
      </b>
       
    </label>
    <input
      autoCapitalize="none"
      autoCorrect="off"
      id="loginPassword"
      name="loginPassword"
      required={true}
      type="password"
    />
    <br />
    <button
      id="admin-login-button"
      type="submit"
    >
      Log In
    </button>
  </form>
</div>
`;
